x/harvest/keeper/interest_test.go

```diff
@@ -0,0 +1,791 @@
+	"github.com/tendermint/tendermint/crypto"
+	tmtime "github.com/tendermint/tendermint/types/time"
+
+	"github.com/kava-labs/kava/app"
+	"github.com/kava-labs/kava/x/harvest"
+	"github.com/kava-labs/kava/x/harvest/types"
+	"github.com/kava-labs/kava/x/pricefeed"
+)
+
+type InterestTestSuite struct {
+	suite.Suite
+}
+
+func (suite *InterestTestSuite) TestCalculateUtilizationRatio() {
+	type args struct {
+		cash          sdk.Dec
+		borrows       sdk.Dec
+		reserves      sdk.Dec
+		expectedValue sdk.Dec
+	}
+
+	type test struct {
+		name string
+		args args
+	}
+
+	testCases := []test{
+		{
+			"normal",
+			args{
+				cash:          sdk.MustNewDecFromStr("1000"),
+				borrows:       sdk.MustNewDecFromStr("5000"),
+				reserves:      sdk.MustNewDecFromStr("100"),
+				expectedValue: sdk.MustNewDecFromStr("0.847457627118644068"),
+			},
+		},
+		{
+			"high util ratio",
+			args{
+				cash:          sdk.MustNewDecFromStr("1000"),
+				borrows:       sdk.MustNewDecFromStr("250000"),
+				reserves:      sdk.MustNewDecFromStr("100"),
+				expectedValue: sdk.MustNewDecFromStr("0.996412913511359107"),
+			},
+		},
+		{
+			"very high util ratio",
+			args{
+				cash:          sdk.MustNewDecFromStr("1000"),
+				borrows:       sdk.MustNewDecFromStr("250000000000"),
+				reserves:      sdk.MustNewDecFromStr("100"),
+				expectedValue: sdk.MustNewDecFromStr("0.999999996400000013"),
+			},
+		},
+		{
+			"low util ratio",
+			args{
+				cash:          sdk.MustNewDecFromStr("1000"),
+				borrows:       sdk.MustNewDecFromStr("50"),
+				reserves:      sdk.MustNewDecFromStr("100"),
+				expectedValue: sdk.MustNewDecFromStr("0.052631578947368421"),
+			},
+		},
+		{
+			"very low util ratio",
+			args{
+				cash:          sdk.MustNewDecFromStr("10000000"),
+				borrows:       sdk.MustNewDecFromStr("50"),
+				reserves:      sdk.MustNewDecFromStr("100"),
+				expectedValue: sdk.MustNewDecFromStr("0.000005000025000125"),
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		suite.Run(tc.name, func() {
+			utilRatio := harvest.CalculateUtilizationRatio(tc.args.cash, tc.args.borrows, tc.args.reserves)
+			suite.Require().Equal(tc.args.expectedValue, utilRatio)
+		})
+	}
+}
+
+func (suite *InterestTestSuite) TestCalculateBorrowRate() {
+	type args struct {
+		cash          sdk.Dec
+		borrows       sdk.Dec
+		reserves      sdk.Dec
+		model         types.InterestRateModel
+		expectedValue sdk.Dec
+	}
+
+	type test struct {
+		name string
+		args args
+	}
+
+	// Normal model has:
+	// 	- BaseRateAPY:      0.0
+	// 	- BaseMultiplier:   0.1
+	// 	- Kink:             0.8
+	// 	- JumpMultiplier:   0.5
+	normalModel := types.NewInterestRateModel(sdk.MustNewDecFromStr("0"), sdk.MustNewDecFromStr("0.1"), sdk.MustNewDecFromStr("0.8"), sdk.MustNewDecFromStr("0.5"))
+
+	testCases := []test{
+		{
+			"normal no jump",
+			args{
+				cash:          sdk.MustNewDecFromStr("1000"),
+				borrows:       sdk.MustNewDecFromStr("3000"),
+				reserves:      sdk.MustNewDecFromStr("5000"),
```

Comment:
This still hits the jump (utilization is 1). Use something like
```
cash: 5000
borrows: 1000
reserves: 1000
```

Reply:
Updated in https://github.com/Kava-Labs/kava/pull/720/commits/60616494ba7b8951f2f05c83e3208f9c7b8f3e3e